File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/notifications/TimedNotificationAlarmReceiver.java
package com.github.fredrik9000.todolist.notifications;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class TimedNotificationAlarmReceiver extends BroadcastReceiver {

    public static final String TODO_DESCRIPTION = "TODO_DESCRIPTION";
    public static final String NOTIFICATION_ID = "NOTIFICATION_ID";

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent timedNotificationJobIntentServiceIntent = new Intent(context, TimedNotificationJobIntentService.class);
        timedNotificationJobIntentServiceIntent.putExtra(NOTIFICATION_ID, intent.getIntExtra(NOTIFICATION_ID, 0));
        TimedNotificationJobIntentService.enqueueWork(context, timedNotificationJobIntentServiceIntent);
    }
}

File: TodoList/app/src/main/java/com/github/fredrik9000/todolist/notifications/TimedNotificationAlarmReceiver.kt
package com.github.fredrik9000.todolist.notifications

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class TimedNotificationAlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        with(Intent(context, TimedNotificationJobIntentService::class.java)) {
            this.putExtra(NOTIFICATION_ID, intent.getIntExtra(NOTIFICATION_ID, 0))
            TimedNotificationJobIntentService.enqueueWork(context, this)
        }
    }

    companion object {
        const val NOTIFICATION_ID = "NOTIFICATION_ID"
    }
}